readers/source_reader.js

```diff
@@ -0,0 +1,31 @@
+"use strict";
+
+const fs        = require('fs-extra');
+
+/*** Synchronous Version ***/
```

Comment:
I left the Async and Sync versions of this function. We can use the async version, but with the memory management required it may be better to just use this synchronous version instead. Let me know what you think.